### ISSUE TITLE
One same EFS cannot be mounted to two different Clusters

### ISSUE THREAD

**Issue by xuanus** [aws/aws-parallelcluster#4062]
- AWS ParallelCluster version: 3.1.4
 - Full cluster configuration without any credentials or personal data.

```
Region: us-west-2
Image:
  Os: ubuntu2004
HeadNode:
  InstanceType: t2.micro 
  Networking:
    SubnetId: subnet-00000HIDE
    ElasticIp: false
  DisableSimultaneousMultithreading: true
  Ssh:
    KeyName: TESTKeyPair
    AllowedIps: 0.0.0.0/0
Scheduling:
  Scheduler: slurm
  SlurmSettings:
    ScaledownIdletime: 1
  SlurmQueues:
    - Name: queue1
      ComputeResources:
        - Name: t2micro
          InstanceType: t2.micro
          MinCount: 0
          MaxCount: 10
          DisableSimultaneousMultithreading: true
      Networking:
        SubnetIds:
          - subnet-000000HIDE
SharedStorage:
  - MountDir: shared
    Name: shared
    StorageType: Efs
    EfsSettings:
      FileSystemId: fs-0611dHIDE  # use cli to create, don't use console to create, otherwise it will not be mounted.
```

 - Cluster name: cluster-fail 
 - Output of `pcluster describe-cluster` command.

pcluster describe-cluster --cluster-name cluster-fail
```
{
  "creationTime": "2022-05-28T06:08:41.207Z",
  "headNode": {
    "launchTime": "2022-05-28T06:11:32.000Z",
    "instanceId": "i-0bf336afHIDE",
    "publicIpAddress": "18.HIDE.213.HIDE",
    "instanceType": "t2.micro",
    "state": "running",
    "privateIpAddress": "172.31.3.33"
  },
  "version": "3.1.4",
  "clusterConfiguration": {
    "url": "https://parallelcluster-fecdc443ef692091-v1-do-not-delete.s3.us-west-2.amazonaws.com/parallelcluster/3.1.4/clusters/cluster-fail-q7j5t8xgmaqd3anz/configs/cluster-config.yaml?versionId=cF7_t43Kikufy_HIDE&AWSAccessKeyId=HIDE&Signature=HIDE%3D&Expires=1653793856"
  },
  "tags": [
    {
      "value": "3.1.4",
      "key": "parallelcluster:version"
    }
  ],
  "cloudFormationStackStatus": "CREATE_FAILED",
  "clusterName": "cluster-fail",
  "computeFleetStatus": "UNKNOWN",
  "cloudformationStackArn": "arn:aws:cloudformation:us-west-2:HIDE:stack/cluster-fail/9f16e000-de4c-11ec-8e17-026000db433f",
  "lastUpdatedTime": "2022-05-28T06:08:41.207Z",
  "region": "us-west-2",
  "clusterStatus": "CREATE_FAILED"
}
```

**Bug description and how to reproduce:**

**1. run CLI "aws efs create-file-system --encrypted --creation-token xxx --region us-west-2"
2. update KeyName, SubnetId and FileSystemId in the cluster configuration
3. "pcluster create-cluster --cluster-name cluster-okay --cluster-configuration my-cluster-config.yaml --rollback-on-failure false"
4. the cluster-okay should be okay
5. "pcluster create-cluster --cluster-name cluster-fail --cluster-configuration my-cluster-config.yaml --rollback-on-failure false"
6. the cluster-fail will fail after 10 attempts to mount EFS**

Following [this guide](https://docs.aws.amazon.com/parallelcluster/latest/ug/troubleshooting-v3.html#troubleshooting-v3-get-logs) and attach the complete cluster log archive with the ticket.

pcluster get-cluster-stack-events --cluster-name cluster-fail --region us-west-2 --query 'events[?resourceStatus==`CREATE_FAILED`]'
```
[
  {
    "eventId": "fd7adbc0-de50-11ec-8edf-0235d00e2fcb",
    "physicalResourceId": "arn:aws:cloudformation:us-west-2:HIDE:stack/cluster-fail/9f16e000-de4c-11ec-8e17-026000db433f",
    "resourceStatus": "CREATE_FAILED",
    "resourceStatusReason": "The following resource(s) failed to create: [HeadNodeWaitCondition20220528060836]. ",
    "stackId": "arn:aws:cloudformation:us-west-2:HIDE:stack/cluster-fail/9f16e000-de4c-11ec-8e17-026000db433f",
    "stackName": "cluster-fail",
    "logicalResourceId": "cluster-fail",
    "resourceType": "AWS::CloudFormation::Stack",
    "timestamp": "2022-05-28T06:39:57.041Z"
  },
  {
    "eventId": "HeadNodeWaitCondition20220528060836-CREATE_FAILED-2022-05-28T06:39:55.175Z",
    "physicalResourceId": "arn:aws:cloudformation:us-west-2:HIDE:stack/cluster-fail/9f16e000-de4c-11ec-8e17-026000db433f/HeadNodeWaitConditionHandle20220528060836",
    "resourceStatus": "CREATE_FAILED",
    "resourceStatusReason": "WaitCondition timed out. Received 0 conditions when expecting 1",
    "resourceProperties": "{\"Timeout\":\"1800\",\"Count\":\"1\",\"Handle\":\"https://cloudformation-waitcondition-us-west-2.s3-us-west-2.amazonaws.com/arn%3Aaws%3Acloudformation%3Aus-west-2%3A016957370679%3Astack/cluster-fail/9f16e000-de4c-11ec-8e17-026000db433f/HeadNodeWaitConditionHandle20220528060836?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20220528T060848Z&X-Amz-SignedHeaders=host&X-Amz-Expires=86399&X-Amz-Credential=HIDE%2F20220528%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Signature=HIDE\"}",
    "stackId": "arn:aws:cloudformation:us-west-2:HIDE:stack/cluster-fail/9f16e000-de4c-11ec-8e17-026000db433f",
    "stackName": "cluster-fail",
    "logicalResourceId": "HeadNodeWaitCondition20220528060836",
    "resourceType": "AWS::CloudFormation::WaitCondition",
    "timestamp": "2022-05-28T06:39:55.175Z"
  }
]
```

Please be sure to attach the following logs:
* From Head node:  
`/var/log/cfn-init.log` 
```
2022-05-28 06:13:56,228 [DEBUG] Running command chef
2022-05-28 06:13:56,228 [DEBUG] No test for command chef
2022-05-28 06:39:00,958 [ERROR] Command chef (cinc-client --local-mode --config /etc/chef/client.rb --log_level info --logfile /var/log/chef-
client.log --force-formatter --no-color --chef-zero-port 8889 --json-attributes /etc/chef/dna.json --override-runlist aws-parallelcluster::co
nfig) failed
2022-05-28 06:39:00,959 [DEBUG] Command chef output:
2022-05-28 06:39:00,959 [ERROR] Error encountered during build of chefConfig: Command chef failed
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/construction.py", line 576, in run_config
    CloudFormationCarpenter(config, self._auth_config).build(worklog)
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/construction.py", line 275, in build
    changes['commands'] = CommandTool().apply(
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/command_tool.py", line 127, in apply
    raise ToolError(u"Command %s failed" % name)
cfnbootstrap.construction_errors.ToolError: Command chef failed
2022-05-28 06:39:00,965 [ERROR] -----------------------BUILD FAILED!------------------------
2022-05-28 06:39:00,965 [ERROR] Unhandled exception during build: Command chef failed
Traceback (most recent call last):
  File "/usr/local/bin/cfn-init", line 176, in <module>
    worklog.build(metadata, configSets)
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/construction.py", line 137, in build
    Contractor(metadata).build(configSets, self)
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/construction.py", line 564, in build
    self.run_config(config, worklog)
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/construction.py", line 576, in run_config
    CloudFormationCarpenter(config, self._auth_config).build(worklog)
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/construction.py", line 275, in build
    changes['commands'] = CommandTool().apply(
  File "/usr/local/lib/python3.8/dist-packages/cfnbootstrap/command_tool.py", line 127, in apply
    raise ToolError(u"Command %s failed" % name)
cfnbootstrap.construction_errors.ToolError: Command chef failed
```

`/var/log/chef-client.log`
```
Recipe: aws-parallelcluster-config::efs_mount
  * directory[/shared] action create[2022-05-28T06:14:20+00:00] INFO: Processing directory[/shared] action create (aws-parallelcluster-config
::efs_mount line 28)
[2022-05-28T06:14:20+00:00] INFO: directory[/shared] created directory /shared

    - create new directory /shared[2022-05-28T06:14:20+00:00] INFO: directory[/shared] owner changed to 0
[2022-05-28T06:14:20+00:00] INFO: directory[/shared] group changed to 0
[2022-05-28T06:14:20+00:00] INFO: directory[/shared] mode changed to 1777

    - change mode from '' to '01777'
    - change owner from '' to 'root'
    - change group from '' to 'root'
  * mount[/shared] action mount[2022-05-28T06:14:20+00:00] INFO: Processing mount[/shared] action mount (aws-parallelcluster-config::efs_moun
t line 37)
[2022-05-28T06:16:30+00:00] INFO: Retrying execution of mount[/shared], 9 attempts left
[2022-05-28T06:18:45+00:00] INFO: Retrying execution of mount[/shared], 8 attempts left
[2022-05-28T06:21:00+00:00] INFO: Retrying execution of mount[/shared], 7 attempts left
[2022-05-28T06:23:15+00:00] INFO: Retrying execution of mount[/shared], 6 attempts left
[2022-05-28T06:25:30+00:00] INFO: Retrying execution of mount[/shared], 5 attempts left
[2022-05-28T06:27:44+00:00] INFO: Retrying execution of mount[/shared], 4 attempts left
[2022-05-28T06:29:59+00:00] INFO: Retrying execution of mount[/shared], 3 attempts left
[2022-05-28T06:32:14+00:00] INFO: Retrying execution of mount[/shared], 2 attempts left
[2022-05-28T06:34:29+00:00] INFO: Retrying execution of mount[/shared], 1 attempt left
[2022-05-28T06:36:44+00:00] INFO: Retrying execution of mount[/shared], 0 attempt left


    ================================================================================
    Error executing action `mount` on resource 'mount[/shared]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    Expected process to exit with [0], but received '32'
    ---- Begin output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev",
"fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
    STDOUT:
    STDERR: mount.nfs4: Connection timed out
    ---- End output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "f
s-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
    Ran ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bb
fbb6.efs.us-west-2.amazonaws.com:/", "/shared"] returned 32

    Resource Declaration:
    ---------------------
    # In /etc/chef/local-mode-cache/cache/cookbooks/aws-parallelcluster-config/recipes/efs_mount.rb

     37:   mount efs_shared_dir do
     38:     device "#{node['cluster']['efs_fs_id']}.efs.#{node['cluster']['region']}.#{node['cluster']['aws_domain']}:/"
     39:     fstype 'nfs4'
     40:     options 'nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev'
     41:     dump 0
     42:     pass 0
     43:     action %i(mount enable)
     44:     retries 10
     45:     retry_delay 6
     46:   end
     47:

    Compiled Resource:
    ------------------
    # Declared in /etc/chef/local-mode-cache/cache/cookbooks/aws-parallelcluster-config/recipes/efs_mount.rb:37:in `from_file'

    mount("/shared") do
      action [:mount, :enable]
      default_guard_interpreter :default
      declared_type :mount
      cookbook_name "aws-parallelcluster-config"
      recipe_name "efs_mount"
      device "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/"
      fstype "nfs4"
      options ["nfsvers=4.1", "rsize=1048576", "wsize=1048576", "hard", "timeo=30", "retrans=2", "noresvport", "_netdev"]
      dump 0
      pass 0
      retries 10
      retry_delay 6
      mount_point "/shared"
      supports {:remount=>false}
      fsck_device "-"
    end

    System Info:
    ------------
    chef_version=17.2.29
    platform=ubuntu
    platform_version=20.04
    ruby=ruby 3.0.1p64 (2021-04-05 revision 0fb782ee38) [x86_64-linux]
    program_name=/bin/cinc-client
    executable=/opt/cinc/bin/cinc-client

[2022-05-28T06:38:59+00:00] INFO: Running queued delayed notifications before re-raising exception
[2022-05-28T06:38:59+00:00] INFO: template[/etc/default/nfs-kernel-server] sending restart action to service[nfs-kernel-server] (delayed)
Recipe: aws-parallelcluster-config::nfs
  * service[nfs-kernel-server] action restart[2022-05-28T06:38:59+00:00] INFO: Processing service[nfs-kernel-server] action restart (aws-para
llelcluster-config::nfs line 39)
[2022-05-28T06:39:00+00:00] INFO: service[nfs-kernel-server] restarted

    - restart service service[nfs-kernel-server]

Running handlers:
[2022-05-28T06:39:00+00:00] ERROR: Running exception handlers
Running handlers complete
[2022-05-28T06:39:00+00:00] ERROR: Exception handlers complete
Cinc Client failed. 46 resources updated in 25 minutes 01 seconds
[2022-05-28T06:39:00+00:00] FATAL: Stacktrace dumped to /etc/chef/local-mode-cache/cache/cinc-stacktrace.out
[2022-05-28T06:39:00+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2022-05-28T06:39:00+00:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: mount[/shared] (aws-parallelcluster-config::efs_mount line 37) had a
n error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '32'
---- Begin output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-
0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
STDOUT:
STDERR: mount.nfs4: Connection timed out
---- End output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-06
11de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
Ran ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6
.efs.us-west-2.amazonaws.com:/", "/shared"] returned 32

```

$ sudo cat /etc/chef/local-mode-cache/cache/cinc-stacktrace.out
```
Generated at 2022-05-28 06:39:00 +0000
Mixlib::ShellOut::ShellCommandFailed: mount[/shared] (aws-parallelcluster-config::efs_mount line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '32'
---- Begin output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
STDOUT:
STDERR: mount.nfs4: Connection timed out
---- End output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
Ran ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] returned 32
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout.rb:300:in `invalid!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout.rb:287:in `error!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout/helper.rb:130:in `shell_out_compacted!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout/helper.rb:52:in `shell_out!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider/mount/mount.rb:120:in `mount_fs'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider/mount.rb:41:in `block (2 levels) in <class:Mount>'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/mixin/why_run.rb:51:in `add_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:265:in `converge_by'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider/mount.rb:40:in `block in <class:Mount>'
(eval):2:in `block in action_mount'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:276:in `instance_eval'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:276:in `compile_and_converge_action'
(eval):2:in `action_mount'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:217:in `run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource.rb:600:in `block in run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource.rb:627:in `with_umask'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource.rb:599:in `run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:74:in `run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:108:in `block in run_all_actions'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:108:in `each'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:108:in `run_all_actions'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:132:in `block in converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/resource_list.rb:96:in `block in execute_each_resource'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:114:in `call_iterator_block'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:103:in `iterate'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:54:in `each_with_index'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/resource_list.rb:94:in `execute_each_resource'
/opt/cinc/embedded/lib/ruby/3.0.0/forwardable.rb:238:in `execute_each_resource'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:130:in `converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:687:in `block in converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:682:in `catch'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:682:in `converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:706:in `converge_and_save'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:286:in `run'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:305:in `run_with_graceful_exit_option'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:281:in `block in run_chef_client'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/local_mode.rb:42:in `with_server_connectivity'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:264:in `run_chef_client'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application/base.rb:352:in `run_application'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:67:in `run'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-bin-17.2.29/bin/cinc-client:25:in `<top (required)>'
/bin/cinc-client:143:in `load'
/bin/cinc-client:143:in `<main>'

>>>> Caused by Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '32'
---- Begin output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
STDOUT:
STDERR: mount.nfs4: Connection timed out
---- End output of ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] ----
Ran ["mount", "-t", "nfs4", "-o", "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=30,retrans=2,noresvport,_netdev", "fs-0611de3f010bbfbb6.efs.us-west-2.amazonaws.com:/", "/shared"] returned 32
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout.rb:300:in `invalid!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout.rb:287:in `error!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout/helper.rb:130:in `shell_out_compacted!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/mixlib-shellout-3.2.5/lib/mixlib/shellout/helper.rb:52:in `shell_out!'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider/mount/mount.rb:120:in `mount_fs'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider/mount.rb:41:in `block (2 levels) in <class:Mount>'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/mixin/why_run.rb:51:in `add_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:265:in `converge_by'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider/mount.rb:40:in `block in <class:Mount>'
(eval):2:in `block in action_mount'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:276:in `instance_eval'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:276:in `compile_and_converge_action'
(eval):2:in `action_mount'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/provider.rb:217:in `run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource.rb:600:in `block in run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource.rb:627:in `with_umask'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource.rb:599:in `run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:74:in `run_action'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:108:in `block in run_all_actions'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:108:in `each'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:108:in `run_all_actions'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:132:in `block in converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/resource_list.rb:96:in `block in execute_each_resource'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:114:in `call_iterator_block'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:103:in `iterate'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/stepable_iterator.rb:54:in `each_with_index'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/resource_collection/resource_list.rb:94:in `execute_each_resource'
/opt/cinc/embedded/lib/ruby/3.0.0/forwardable.rb:238:in `execute_each_resource'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/runner.rb:130:in `converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:687:in `block in converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:682:in `catch'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:682:in `converge'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:706:in `converge_and_save'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/client.rb:286:in `run'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:305:in `run_with_graceful_exit_option'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:281:in `block in run_chef_client'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/local_mode.rb:42:in `with_server_connectivity'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:264:in `run_chef_client'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application/base.rb:352:in `run_application'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-17.2.29/lib/chef/application.rb:67:in `run'
/opt/cinc/embedded/lib/ruby/gems/3.0.0/gems/chef-bin-17.2.29/bin/cinc-client:25:in `<top (required)>'
/bin/cinc-client:143:in `load'
```

**Comment by hanwen-pcluste** [comment]
Hello Xuanus,

Sorry for the late reply! Thank you for the detailed information.

When mounting an existing EFS, please make sure the security groups of the mount targets of the EFS are properly configured. This is [EFS guide](https://docs.aws.amazon.com/efs/latest/ug/accessing-fs-create-security-groups.html).


### The difference between creating a EFS through CLI vs AWS web console: 
* When creating a EFS through CLI, only the file system is created. Then when ParallelCluster creates a cluster using the EFS, ParallelClsuter creates mount targets with the correct security group for the file system. However, the security group of the mount targets cannot work with a second cluster, because ParallelCluster tries to allow the minimal connections.
* When creating a EFS through AWS web console, the mount targets are created together with the file system. The mount targets use the default security group of the VPC. Because the default security group does not allow connections though NFS, ParallelCluster cannot mount the EFS.

### How to solve the problem
The following actions are required if you create the EFS from console or try to mount the EFS to more than one cluster.
In the AWS console EFS configuration page: Networking -> Manage, use a security group that allows connections through port 2049 (NFS) to cluster instances.

Please let us know if you have further question

Thank you,
Hanwen

**Comment by xuanus** [comment]
Hi Hanwen,

Thank you so much for your response! Very helpful!

One follow up question:

In https://docs.aws.amazon.com/parallelcluster/latest/ug/SharedStorage-v3.html
It says "The shared storage settings for the cluster. Up to 5 Amazon EBS, 1 Amazon EFS, and 1 FSx for Lustre file systems can be specified."

May I know why there is a hard limit on 1 Shared EFS? How to mount multiple EFS to one cluster? Any workaround?

**Comment by chenwany** [comment]
Hi Xuanus, we are working on adding multiple EFS support, will let you know once it is supported.
Current'y there's no easy workaround on adding more EFS.

Thanks!

**Comment by chenwany** [comment]
You can track the feature request here: support multiple EFS mounts: https://github.com/aws/aws-parallelcluster/issues/1735
I am going to close this issue soon if there's no further question.

Thanks

**Comment by xuanus** [comment]
@chenwany looks like supporting multiple EFS mounts: https://github.com/aws/aws-parallelcluster/issues/1735 was released in 3.2.0

I am trying to share the same EFS between two PClusters within the same VPC and subnet.
The first PCluster works fine.
The second PCluster got
```
{
      "level": "ERROR",
      "type": "EfsIdValidator",
      "message": "There is an existing Mount Target fsmt-xxx in the Availability Zone us-west-2c for EFS fs-xxx, but it does not have a security group that allows inbound and outbound rules to support NFS. Please modify the Mount Target's security group, to allow traffic on port 2049."
}
```


Question:
1. will PCluster support "share the same EFS between two PClusters within the same VPC and subnet" in 3.2.0 or later?
2. To update/add EFS security group to allow connections through port 2049 (NFS) to cluster instances, what is the best practice to set Source? 0.0.0.0/0 is not secure.

**Comment by enrico-usai** [comment]
Hi @xuanus 

1. in ParallelCluster 3.2.0 we added the possibility to attach multiple EFS to the same cluster but not to use the same EFS into different clusters. The key point is described in the validator message. The mount point is already created so ParallelCluster cannot create it, you have to modify the security group associated to the mount target by hand.

2. You can create a new security group and use it as Source for the EFS, then pass it in the [`AdditionalSecurityGroups`](https://docs.aws.amazon.com/parallelcluster/latest/ug/Scheduling-v3.html#yaml-Scheduling-AwsBatchQueues-Networking-AdditionalSecurityGroups) field of both head node and compute Networking of the cluster you'll have to use it. In this way all the nodes of the new cluster will be associated to the new security group that will have access to the EFS.

Let us know if it helps.

Enrico
